relay-server/src/services/store.rs

```diff
@@ -891,6 +908,66 @@ impl StoreService {
         Ok(())
     }
 
+    fn produce_replay_video(
+        &self,
+        event_id: Option<EventId>,
+        scoping: Scoping,
+        payload: Bytes,
+        start_time: Instant,
+        retention: u16,
+    ) -> Result<(), StoreError> {
+        #[derive(Deserialize)]
+        struct VideoEvent<'a> {
+            replay_event: &'a [u8],
+            replay_recording: &'a [u8],
+            replay_video: &'a [u8],
+        }
+
+        let Ok(VideoEvent {
+            replay_video,
+            replay_event,
+            replay_recording,
+        }) = rmp_serde::from_slice::<VideoEvent>(&payload)
+        else {
+            self.outcome_aggregator.send(TrackOutcome {
+                category: DataCategory::Replay,
+                event_id,
+                outcome: Outcome::Invalid(DiscardReason::InvalidReplayEvent),
+                quantity: 1,
+                remote_addr: None,
+                scoping,
+                timestamp: instant_to_date_time(start_time),
+            });
+            return Ok(());
+        };
+
+        self.produce_replay_event(
+            event_id.ok_or(StoreError::NoEventId)?,
+            scoping.organization_id,
+            scoping.project_id,
+            start_time,
+            retention,
+            Bytes::copy_from_slice(replay_event),
+        )?;
```

Comment:
Why do we need to produce the event a second time? It's already published as part of the replay recording, right?

Reply:
Its being published to a different consumer.  We're producing one event to the replay_recording consumer and another event to the replay_event consumer.  The duplicated data being sent to the recording consumer for reasons other than storing the replay_event bytes.  Refer to this PR https://github.com/getsentry/relay/pull/3035 where we merged the payloads.

In the future we will drop the `produce_replay_event` function from Relay but we're not ready yet.